master/buildbot/test/fake/httpclientservice.py

```diff
@@ -113,7 +113,7 @@ def assertNoOutstanding(self):
         self.case.assertEqual(0, len(self._expected),
                               "expected more http requests:\n {!r}".format(self._expected))
 
-    def _doRequest(self, method, ep, params=None, data=None, json=None):
+    def _doRequest(self, method, ep, params=None, data=None, json=None, timeout=None):
```

Comment:
if you are adding a requests based option here, you need to make sure it will also work with treq too (with unit test and integration tests proving that it works).
I think this belongs to another improvement PR